### PR TITLE
feat: lambda bundling with cdk-assets

### PR DESCRIPTION
**Pull request by Hi-Fi**
Currently there's no code generating asset manifest that would be using bundling. That generation is easier to build when there's some verified way to make bundling with cdk-assets.

- Relates to at least following issues:
    -  #19157
    - #21848
----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk-assets/lib/private/handlers/bundlable-files.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    const bundler = new ContainerBunder(
+      this.asset.source.bundling,
+      path.resolve(this.workDir, this.asset.source.path),
+    );
+
+    const bundledPath = await bundler.bundle();
+
+    // Hack to get things tested
+    (this.asset.source.path as any) = bundledPath;
```
Comment (Hi-Fi) [inline packages/cdk-assets/lib/private/handlers/bundlable-files.ts:24]: Currently added this hack to test things, but how this should be done?
1. add output path to assembly and pre-generate it when generating assemby?
2. modify publish part to allow overriding path with argument?

**Review by corymhall — CHANGES_REQUESTED** [review]
@Hi-Fi thanks for taking the time to create this PR! Since this is labeled as a `p2` _and_ is adding some pretty significant functionality we will most likely not be able to review this without an RFC. I think there are a lot of ideas around how to enhance/refactor the assets/bundling functionality and it needs a much bigger discussion than is appropriate for a PR review.

**Comment by Hi-Fi** [comment]
> @Hi-Fi thanks for taking the time to create this PR! Since this is labeled as a `p2` _and_ is adding some pretty significant functionality we will most likely not be able to review this without an RFC. I think there are a lot of ideas around how to enhance/refactor the assets/bundling functionality and it needs a much bigger discussion than is appropriate for a PR review.

Just interested, how that p2 got here? Both referred issues are p1.

**Comment by Hi-Fi** [comment]
@corymhall was there something to improve in this PR, or was it just waiting auto-closure die inactivity after change to draft?

**Pull request closed without merging**